core/src/main/java/ai/timefold/solver/core/impl/score/stream/collector/connected_ranges/ConnectedRangeChainImpl.java

```diff
@@ -235,20 +239,39 @@ void removeInterval(Interval<Interval_, Point_> interval) {
     }
 
     @Override
-    public Iterable<IntervalCluster<Interval_, Point_, Difference_>> getIntervalClusters() {
+    public Iterable<ConnectedRange<Interval_, Point_, Difference_>> getConnectedRanges() {
         return (Iterable) clusterStartSplitPointToCluster.values();
     }
 
     @Override
-    public Iterable<IntervalBreak<Interval_, Point_, Difference_>> getBreaks() {
+    public Iterable<RangeGap<Point_, Difference_>> getGaps() {
         return (Iterable) clusterStartSplitPointToNextBreak.values();
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (!(o instanceof ConnectedRangeChainImpl<?, ?, ?> that))
+            return false;
+        return Objects.equals(clusterStartSplitPointToCluster,
+                that.clusterStartSplitPointToCluster)
+                && Objects.equals(splitPointSet,
+                        that.splitPointSet)
+                && Objects.equals(clusterStartSplitPointToNextBreak,
+                        that.clusterStartSplitPointToNextBreak);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(clusterStartSplitPointToCluster, splitPointSet, clusterStartSplitPointToNextBreak);
+    }
+
     @Override
     public String toString() {
         return "Clusters {" +
-                "intervalClusters=" + getIntervalClusters() +
-                ", breaks=" + getBreaks() +
+                "intervalClusters=" + getConnectedRanges() +
+                ", breaks=" + getGaps() +
                 '}';
```

Comment:
Please fix naming. Possibly in the entire file; speaks of clusters, but there no longer is any such class.